unit-tests-java8/src/test/java/org/eclipse/collections/test/map/mutable/UnmodifiableHashMapTest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2024 Goldman Sachs and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v. 1.0 which accompany this distribution.
+ * The Eclipse Public License is available at http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ */
+
+package org.eclipse.collections.test.map.mutable;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+
+import org.eclipse.collections.impl.test.junit.Java8Runner;
+import org.eclipse.collections.test.map.UnmodifiableMapTestCase;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.fail;
+
+@RunWith(Java8Runner.class)
+public class UnmodifiableHashMapTest
+        implements UnmodifiableMapTestCase
+{
+    private static final long CURRENT_TIME_MILLIS = System.currentTimeMillis();
+
+    @Override
+    public <T> Map<Object, T> newWith(T... elements)
+    {
+        Random random = new Random(CURRENT_TIME_MILLIS);
+        Map<Object, T> result = new HashMap<>();
+        for (T each : elements)
+        {
+            assertNull(result.put(random.nextDouble(), each));
+        }
+        return Collections.unmodifiableMap(result);
```

Comment:
This is an important test to ensure consistency across implementations. This runs the same assertions against Collections.unmodifiableMap(new HashMap<>())